lib/pdk/cli/release.rb

```diff
@@ -0,0 +1,295 @@
+        end
+      end
+
+      PDK::CLI::Util.analytics_screen_view('release', opts)
+
+      PDK::CLI::Util.validate_puppet_version_opts(opts)
+
+      PDK::CLI::Util.module_version_check
+
+      report = PDK::Report.new
+      puppet_env = PDK::CLI::Util.puppet_from_opts_or_env(opts)
+      PDK::Util::PuppetVersion.fetch_puppet_dev if opts[:'puppet-dev']
+      PDK::Util::RubyVersion.use(puppet_env[:ruby_version])
+
+      opts.merge!(puppet_env[:gemset])
+
+      PDK::Util::Bundler.ensure_bundle!(puppet_env[:gemset])
+
+      if opts[:skip_validation]
+        PDK.logger.info _('Skipping module validation')
+      else
+        exit_code = 0
+        validators = PDK::Validate.validators
+        validators.each do |validator|
+          validator_exit_code = validator.invoke(report, opts.dup)
+          exit_code = validator_exit_code if validator_exit_code != 0
+        end
+
+        exit exit_code if exit_code != 0
+      end
+
+      module_metadata = PDK::Module::Metadata.from_file('metadata.json')
+
+      PDK.logger.info _('Releasing %{module_name} - from version %{module_version}') % {
+        module_name:    module_metadata.data['name'],
+        module_version: module_metadata.data['version'],
+      }
+
+      if opts[:skip_changelog]
+        PDK.logger.info _('Skipping automatic changelog generation')
+        if opts[:version_number]
+          PDK.logger.info _('Updating version to %{module_version}') % {
+            module_version: opts[:version_number],
+          }
+
+          # Bump version in metadata file
+          module_metadata.data['version'] = opts[:version_number]
+          module_metadata.write!('metadata.json')
+        end
+      else
+        new_version = 'v0.0.1'
+        changelog_generator = PDK::CLI::Util::ChangelogGenerator.new
+        if opts[:version_number]
+          new_version = opts[:version_number]
+        else
+          # Use changelog generator to establish version
+          PDK.logger.info _('Generating changelog to get next version number')
+          changelog_generator.generate_changelog
+
+          new_version = changelog_generator.get_next_version(module_metadata.data['version'])
+        end
+
+        PDK.logger.info _('Updating version to %{module_version}') % {
+          module_version: new_version,
+        }
+
+        # Bump version in metadata file
+
+        module_metadata.data['version'] = new_version
+        module_metadata.write!('metadata.json')
+
+        # Create new changelog with the correct version
+        changelog_generator.generate_changelog
+        PDK.logger.info _("New changelog generated with version #{new_version}")
+      end
+
+      if opts[:skip_documentation]
+        PDK.logger.info _('Skipping documentation update for this module')
+      else
+        PDK.logger.info _('Updating documentation using puppet strings')
+        result = system('bundle exec puppet strings generate --format markdown --out REFERENCE.md')
```

Comment:
This won't work in the packaged versions of PDK.  See `lib/pdk/cli/bundle.rb` for more examples as to how to run a bundle commend in the context of a module.

Reply:
Fixed, thanks